### ISSUE TITLE
Mutators for C# 9 Pattern Matching

### ISSUE THREAD

**Issue by hannahchan** [stryker-mutator/stryker-net#1338]
I'm only just learning about this now so this might not make sense. I've got code like this;

```C#
if (myInput is not null)
{
    doSomething();
}
```
The `is not` pattern combination is new to C# 9 and Stryker is currently unable to mutate it. It looks like there are other pattern operators such as `and` and/or `or` 😄 . It would be nice to have mutators for those as well.

Someone will need to take a look at the pattern matching specification of C# 9. The following might help.

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/patterns3

**Comment by Mobrockers** [comment]
Pattern matching mutators would be awesome :)

**Comment by richardwerkman** [comment]
I think there is not much else than `not`, `and` and `or`. So the mutator would look like:

## Pattern matching 
| Original | Mutated | 
| ------------- | ------------- | 
| `is` | `is not` |
| `is not` | `is` |
| `and` | `or` |
| `or` | `and` |

We could also remove brackets like:
``` cs
bool IsLetter(char c) => c is (>= 'a' and <= 'z') or (>= 'A' and <= 'Z');
```

``` cs
bool IsLetter(char c) => c is >= 'a' and <= 'z' or >= 'A' and <= 'Z';
```

**Comment by Mobrockers** [comment]
Those are the interesting new operators to mutate, but I don't think we're currently mutating all existing pattern matching values since sometimes they need to be static. There might also be some fun stuff we could do with the is Type pattern matching. For example change an is Type pattern to a is var pattern.

**Comment by rouke-broersma** [comment]
Implemented by #2313 for switch patterns
